### PR TITLE
Add Rector configuration

### PR DESCRIPTION
**Pull request by jordisala1991**
## Subject

I am targeting this branch, because this is BC.

### CODE REVIEW

**Comment by jordisala1991** [comment]
To me it looks like pedantic since it doesn't modify anything on the user side.

File: composer.json [hunk 1]
```diff
@@ -55,6 +55,7 @@
         "phpunit/phpunit": "^9.5.13",
         "psalm/plugin-phpunit": "^0.16",
         "psalm/plugin-symfony": "^3.0",
+        "rector/rector": "dev-main",
```
Comment (jordisala1991) [inline composer.json]: I would like to use ^0.12 again, but this needs to be fixed first: https://github.com/rectorphp/rector/issues/7141

Comment (jordisala1991) [reply composer.json]:
```suggestion
        "rector/rector": "^0.12",
```

To apply when the above is fixed.

**Comment by jordisala1991** [comment]
Added dev-kit: https://github.com/sonata-project/dev-kit/pull/2008

**Review by VincentLanglet — DISMISSED** [review]

**Pull request merged**